File: java/ch17-04-arithmetic/src/main/java/com/thesoftwaregorilla/tdd/money/IExpression.java
package com.thesoftwaregorilla.tdd.money;

import java.math.BigDecimal;

public interface IExpression<T extends ICurrencyHolder<T>> {
    public T add(T addend);
    public T subtract(T subtrahend);
    public T multiply(BigDecimal multiplier);
    public T divide(BigDecimal divisor);
}

File: cs/ch17-04-arithmetic/Money/IExpression.cs
namespace TheSoftwareGorilla.TDD.Money;

public interface IExpression<T> where T : ICurrencyHolder<T>
{
    public T Add(Money addend);
    public T Subtract(Money subtrahend);
    public T Multiply(decimal multiplier);
    public T Divide(decimal divisor);
}
